packages/flutter_tools/templates/module/ios/library/Flutter.tmpl/podhelper.rb

```diff
@@ -1,75 +1,112 @@
-framework_dir = File.join(flutter_application_path, '.ios', 'Flutter')
+# Install pods needed to embed Flutter and plugins
+# from the Flutter module Podfile.
+#
+# target 'Runner' do
+#   install_flutter_plugin_pods 'my_flutter'
+# end
+def install_flutter_plugin_pods(flutter_application_path = nil)
+  # If this wasn't specified, assume it's two levels up from the directory of this script.
+  flutter_application_path ||= File.join('..', '..')
 
-engine_dir = File.join(framework_dir, 'engine')
-if !File.exist?(engine_dir)
+  engine_dir = File.join(__dir__, 'engine')
+  if !File.exist?(engine_dir)
     # Copy the debug engine to have something to link against if the xcode backend script has not run yet.
-    debug_framework_dir = File.join(flutter_root(flutter_application_path), 'bin', 'cache', 'artifacts', 'engine', 'ios')
+    # CocoaPods will not generate a install framework command if the vendored framework does not exist.
+    debug_framework_dir = File.join(flutter_root, 'bin', 'cache', 'artifacts', 'engine', 'ios')
     FileUtils.mkdir_p(engine_dir)
     FileUtils.cp_r(File.join(debug_framework_dir, 'Flutter.framework'), engine_dir)
     FileUtils.cp(File.join(debug_framework_dir, 'Flutter.podspec'), engine_dir)
-end
+  end
 
-pod 'Flutter', :path => engine_dir
-pod 'FlutterPluginRegistrant', :path => File.join(framework_dir, 'FlutterPluginRegistrant')
+  pod 'Flutter', :path => engine_dir, :inhibit_warnings => true
+  pod 'FlutterPluginRegistrant', :path => File.join(__dir__, 'FlutterPluginRegistrant'), :inhibit_warnings => true
 
-symlinks_dir = File.join(framework_dir, '.symlinks')
-FileUtils.mkdir_p(symlinks_dir)
-plugin_pods = parse_KV_file(File.join(flutter_application_path, '.flutter-plugins'))
-plugin_pods.map { |r|
+  symlinks_dir = File.join(__dir__, '.symlinks')
+  FileUtils.mkdir_p(symlinks_dir)
+  plugin_pods = parse_KV_file(File.join(flutter_application_path, '.flutter-plugins'))
+  plugin_pods.map do |r|
     symlink = File.join(symlinks_dir, r[:name])
     FileUtils.rm_f(symlink)
     File.symlink(r[:path], symlink)
-    pod r[:name], :path => File.join(symlink, 'ios')
-}
+    pod r[:name], :path => File.join(symlink, 'ios'), :inhibit_warnings => true
+  end
+end
+
+# Install pod needed to embed Flutter application.
+def install_flutter_application_pod(flutter_application_path, flutter_application_name = nil)
+  flutter_application_name ||= File.basename(flutter_application_path)
+
+  app_framework_dir = File.join(__dir__, 'App.framework')
+  app_framework_dylib = File.join(app_framework_dir, 'App')
+  if !File.exist?(app_framework_dylib)
+    # Fake an App.framework to have something to link against if the xcode backend script has not run yet.
+    # CocoaPods will not generate a install framework command if the vendored framework binary does not exist.
+    FileUtils.mkdir_p(app_framework_dir)
+    `echo "static const int Moo = 88;" | xcrun clang -x c -dynamiclib -o "#{app_framework_dylib}" -`
```

Comment:
Matches the hack in xcode_backend.

Reply:
😂TIL... that's wild

Reply:
I actually came up with an almost-identical hack independently until I saw the one in xcode_backend.  CocoaPods just will not add the command to install a framework unless there's a dylib present at pod install, even before the project can be built.